contrib/ros/README.md

```diff
@@ -0,0 +1,138 @@
+# rllab.contrib.ros Instructions
+## Introduction
+Most of the current environments for reinfocement learning training(e.g. gym's environments) are not particularly 
+designed for robot learning. Consequently, most of their environments are not for continuous action space task. And 
+it is also hard for robot learning researchers to use the same environment interface when they want to test their 
+algorithms on real robots. 
+
+rllab.contrib.ros's environments take a different approach. It's targeting at provide the same interfaces for both of the 
+training on real and simulation robots. It's developed by Hejia Zhang.
+## Support Platform
+| Robot        | Workstation OS | ROS    | Python | Simulation Platform | 
+|:------------:|:--------------:| :-----:|:------:|:-------------------:|
+| Sawyer       | Ubuntu16.04    | Kinetic| 3      |Gazebo 7.0           |
+## Installation
+- Assuming we got a clean system which only has rllab running.
+- Check if our robots and workstations are supported by rllab.contrib.ros    
+- Following is how we setup the rllab.contrib.ros's environment for sawyer robot
+### Setup Workstation
+#### INSTALL ROS
+##### Configure Ubuntu repositories
+Configure your Ubuntu repositories to allow "restricted," "universe," and "multiverse." **Likely, they are already configured properly, and you only need to confirm the configuration.**
+##### Setup your sources.list
+    $ sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu xenial main" > /etc/apt/sources.list.d/ros-latest.list'
+##### Setup your keys
+    $ sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+##### Update to Latest Software Lists
+    $ sudo apt-get update
+##### Install ROS Kinetic Desktop Full
+    $ sudo apt-get install ros-kinetic-desktop-full
+##### Initialize rosdep
+rosdep enables you to easily install system dependencies for source you want to compile and is required to run some core components in ROS.
+    
+    $ sudo rosdep init
+    $ rosdep update
+##### Install rosinstall
+    $ sudo apt-get install python-rosinstall
+#### CREATE DEVELOPMENT WORKSPACE
+##### Create ROS Workspace
+    $ mkdir -p ~/<YOUR_FAVORITE_WORKSPACE_NAME>/src
+##### Source ROS Setup
+    $ source /opt/ros/kinetic/setup.bash
+You can also add this to you .bashrc by using
+    
+    $ echo "source /opt/ros/kinetic/setup.bash" >> ~/.bashrc
+##### Build
+    $ cd ~/<YOUR_FAVORITE_WORKSPACE_NAME>
+    $ catkin_make
```

Comment:
Base path: /home/sawyer/ros_ws_test
Source space: /home/sawyer/ros_ws_test/src
Build space: /home/sawyer/ros_ws_test/build
Devel space: /home/sawyer/ros_ws_test/devel
Install space: /home/sawyer/ros_ws_test/install
Creating symlink "/home/sawyer/ros_ws_test/src/CMakeLists.txt" pointing to "/opt/ros/kinetic/share/catkin/cmake/toplevel.cmake"
####
#### Running command: "cmake /home/sawyer/ros_ws_test/src -DCATKIN_DEVEL_PREFIX=/home/sawyer/ros_ws_test/devel -DCMAKE_INSTALL_PREFIX=/home/sawyer/ros_ws_test/install -G Unix Makefiles" in "/home/sawyer/ros_ws_test/build"
####
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using CATKIN_DEVEL_PREFIX: /home/sawyer/ros_ws_test/devel
-- Using CMAKE_PREFIX_PATH: /opt/ros/kinetic
-- This workspace overlays: /opt/ros/kinetic
-- Found PythonInterp: /home/sawyer/anaconda2/bin/python (found version "2.7.14") 
-- Using PYTHON_EXECUTABLE: /home/sawyer/anaconda2/bin/python
-- Using Debian Python package layout
-- Using empy: /usr/bin/empy
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/sawyer/ros_ws_test/build/test_results
-- Found gmock sources under '/usr/src/gmock': gmock will be built
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found gtest sources under '/usr/src/gmock': gtests will be built
-- Using Python nosetests: /usr/bin/nosetests-2.7
ImportError: "from catkin_pkg.package import parse_package" failed: No module named catkin_pkg.package
Make sure that you have installed "catkin_pkg", it is up to date and on the PYTHONPATH.
CMake Error at /opt/ros/kinetic/share/catkin/cmake/safe_execute_process.cmake:11 (message):
  execute_process(/home/sawyer/anaconda2/bin/python
  "/opt/ros/kinetic/share/catkin/cmake/parse_package_xml.py"
  "/opt/ros/kinetic/share/catkin/cmake/../package.xml"
  "/home/sawyer/ros_ws_test/build/catkin/catkin_generated/version/package.cmake")
  returned error code 1
Call Stack (most recent call first):
  /opt/ros/kinetic/share/catkin/cmake/catkin_package_xml.cmake:74 (safe_execute_process)
  /opt/ros/kinetic/share/catkin/cmake/all.cmake:151 (_catkin_package_xml)
  /opt/ros/kinetic/share/catkin/cmake/catkinConfig.cmake:20 (include)
  CMakeLists.txt:52 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/sawyer/ros_ws_test/build/CMakeFiles/CMakeOutput.log".
See also "/home/sawyer/ros_ws_test/build/CMakeFiles/CMakeError.log".
Invoking "cmake" failed

Reply:
catkin_make -DPYTHON_EXECUTABLE:FILEPATH=/home/sawyer/anaconda2/envs/rllab3/bin/python